cmd/user-agent/src/pages/chapi/wallet/didcomm/mediator.js

```diff
@@ -96,9 +102,34 @@ export class AgentMediator {
         return response.invitation
     }
 
-    // TODO to be implemented
-    async requestDID() {
-        console.warn('feature for requesting peer DID from router is not yet available')
+    async requestDID(reqDoc) {
+        let res = await this.aries.mediator.getConnections().catch(err => {
+            if (!err.message.includes("router not registered")) {
+                throw err
+            }
+        })
+
+        if (!res.connections || res.connections.length == 0) {
+            console.error('failed to send connection request to router, no connection found !')
+            throw 'could not find connection with router'
+        }
+
+
+        let response = await this.messenger.sendAndWaitForReply(res.connections[0], {
```

Comment:
@sudeshrshetty how do you know the first router in this list is the right one?

This should be randomized at minimum.

cc @soluchok

Reply:
`await getMediatorConnections(this.aries, true)` instead of `res.connections[0]`

Reply:
created a task for this one, need thorough discussion.

Reply:
ignore my previous comment, will fix it. overlooked that function while resolving conflicts.

Reply:
fixed